src/lfs/application.lua

```diff
@@ -45,42 +70,61 @@ if #dht_sensors > 0 then
   for i, sensor in pairs(dht_sensors) do
     local pollInterval = tonumber(sensor.poll_interval) or 0
     pollInterval = (pollInterval > 0 and pollInterval or 3) * 60 * 1000
-    print("Heap:", node.heap(), "Polling DHT on pin " .. sensor.pin .. " every " .. pollInterval .. "ms")
-    tmr.create():alarm(pollInterval, tmr.ALARM_AUTO, function() readDht(sensor.pin) end)
-    readDht(sensor.pin)
+    if sensor.zone ~= nil then
+      print("Heap:", node.heap(), "Polling DHT on zone " .. sensor.zone .. " every " .. pollInterval .. "ms")
+    else
+      print("Heap:", node.heap(), "Polling DHT on pin " .. sensor.pin .. " every " .. pollInterval .. "ms")
+    end
+    tmr.create():alarm(pollInterval, tmr.ALARM_AUTO, function() readDht(sensor) end)
+    readDht(sensor)
   end
 end
 
 -- initialize ds18b20 temp sensors
 if #ds18b20_sensors > 0 then
 
-  local function ds18b20Callback(pin)
+  local function ds18b20Callback(sensor)
     local callbackFn = function(temps)
       for addr,value in pairs(temps) do
         print("Heap:", node.heap(), "Temperature:", value)
-        table.insert(sensorPut, { pin = pin, temp = value,
-          addr = string.format(('%02X:%02X:%02X:%02X:%02X:%02X:%02X:%02X'):format(addr:byte(1,8)))})
+        if sensor.zone ~= nil then
+          table.insert(sensorPut, { zone = sensor.zone, temp = value,
+            addr = string.format(('%02X:%02X:%02X:%02X:%02X:%02X:%02X:%02X'):format(addr:byte(1,8)))})
```

Comment:
maybe extract the format string to reduce copypasta

Reply:
Good call, will do.